### PR TITLE
planner, runtimefilter: resolve column index for table scan

### LINKED ISSUE #44450: Semi join error result when runtime filter mode is LOCAL
## Bug Report
Query:
```select xxx from xxx where xxx in (xxx) ```
Result error when ```set tidb_runtime_filter_mode=LOCAL;```

### 1. Minimal reproduce step (Required)
1. create table and insert values
```
create table t1(k1 int, t1_tinyint tinyint, t1_smallint smallint, t2_mediumint mediumint, t1_bigint bigint, t1_int int, t1_bit bit,  t1_bool bool, t1_float float, t1_double DOUBLE, t1_decimal decimal(10, 2));
insert into t1 values (1,1,1,1,1111111,1, 1, 1, 1.0, 1.00, 1.0);
insert into t1 values (1,1,1,1,1111111,1, 1, 1, 1.0, 1.00, 1.0);
create table t2(k1 int, t2_tinyint tinyint, t2_smallint smallint, t2_mediumint mediumint, t2_bigint bigint, t2_int int, t2_bit bit,  t2_bool bool, t2_float float, t2_double DOUBLE, t2_decimal decimal(10, 2));
insert into t2 values (1,1,1,1,1111111,1, 1, 1, 1.0, 1.00, 1.0);
insert into t2 values (1,1,1,1,1111111,1, 1, 1, 1.0, 1.00, 1.0);
```

2. enable runtime filter and mpp
```
alter table t1 set tiflash replica 1;
alter table t2 set tiflash replica 1;
set tidb_isolation_read_engines="tidb,tiflash";
set tidb_runtime_filter_mode="LOCAL";
```

3. query semi join or in subquery
```
select t1_tinyint from t1 where t1.t1_tinyint in (select t2_tinyint from t2 where t1_bigint=t2_bigint); 
```

4. error result
```
Empty set (0.03 sec)
```

### 2. What did you expect to see? (Required)

```
mysql> select t1_tinyint from t1 where t1.t1_tinyint in (select t2_tinyint from t2 where t1_bigint=t2_bigint);
+------------+
| t1_tinyint |
+------------+
|          1 |
|          1 |
+------------+
```

### 3. What did you see instead (Required)

```
mysql> select t1_tinyint from t1 where t1.t1_tinyint in (select t2_tinyint from
t2 where t1_bigint=t2_bigint);
Empty set (39.41 sec)
```

### 4. What is your TiDB version? (Required)

After v7.1

### PR DESCRIPTION
**Pull request by elsa0520**
The column of table scan needs to be resolved, otherwise its index is wrong. If the index is wrong, the target column of the runtime filter will also be wrong, and finally the runtime filter will be hung on the wrong target column. Lead to wrong results.

### What problem does this PR solve?

Issue Number: close #44450

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by fixdb — APPROVED** [review]
+1

File: planner/core/resolve_indices.go [hunk 1]
```diff
@@ -655,7 +655,19 @@ func (p *PhysicalApply) ResolveIndices() (err error) {
...
+	}
+	return p.ResolveIndicesItself()
+}
+
+// ResolveIndicesItself implements PhysicalTableScan interface.
+func (p *PhysicalTableScan) ResolveIndicesItself() (err error) {
+	for i, column := range p.schema.Columns {
+		column.Index = i
+	}
+	return
```
Comment (chrysan) [inline planner/core/resolve_indices.go:670]: Is it possible to add UT for it?

Comment (elsa0520) [reply planner/core/resolve_indices.go:670]: No, This attribute is mainly used for tiflash, so it cannot be tested by only writing tidb single test.
But after I do a full test, I can see how to add it to the daily regression test.

**Review by winoros — APPROVED** [review]

**Review by chrysan — APPROVED** [review]

**Review by AilinKid — APPROVED** [review]

**Pull request merged**